lib/repl.js

```diff
@@ -1209,31 +1201,22 @@ function complete(line, callback) {
     if (!subdir) {
       completionGroups.push(_builtinLibs);
     }
-
-    completionGroupsLoaded();
   } else if (fsAutoCompleteRE.test(line)) {
-    filter = '';
+    let baseName = '';
     let filePath = line.match(fsAutoCompleteRE)[1];
-    let fileList;
+    let fileList = gracefulReaddir(filePath, { withFileTypes: true });
 
-    try {
-      fileList = fs.readdirSync(filePath, { withFileTypes: true });
-      completionGroups.push(fileList.map((dirent) => dirent.name));
-      completeOn = '';
-    } catch {
-      try {
-        const baseName = path.basename(filePath);
-        filePath = path.dirname(filePath);
-        fileList = fs.readdirSync(filePath, { withFileTypes: true });
-        const filteredValue = fileList.filter((d) =>
-          d.name.startsWith(baseName))
-          .map((d) => d.name);
-        completionGroups.push(filteredValue);
-        completeOn = baseName;
-      } catch {}
+    if (!fileList) {
+      baseName = path.basename(filePath);
+      filePath = path.dirname(filePath);
+      fileList = gracefulReaddir(filePath, { withFileTypes: true }) || [];
```

Comment:
Nullish coalescing operator here and in other places that use || for defaults?
```suggestion
      fileList = gracefulReaddir(filePath, { withFileTypes: true }) ?? [];
```

Reply:
I thought about that but I would like to be able to backport this PR without any code changes.

Reply:
What Node version couldn't use ?? In Node but require the ability to nautocomplete ?. in the repl?

Reply:
I would also backport it to versions that do not support it. It would not change anything on those versions besides preventing conflicts while backporting other commits.